public/app/plugins/datasource/loki/live_streams.ts

```diff
@@ -42,9 +43,26 @@ export class LiveStreams {
         appendResponseToBufferedData(response, data);
         return [data];
       }),
-      catchError(err => {
-        return throwError(`error: ${err.reason}`);
-      }),
+      retryWhen((attempts: Observable<any>) =>
+        attempts.pipe(
+          mergeMap((error, i) => {
+            // Code 1006 is used to indicate that a connection was closed abnormally.
+            // If connection was closed abnormally, we wish to retry, otherwise throw error.
+            const retryAttempt = i + 1;
+            if (error.code === 1006) {
+              if (retryAttempt > 10) {
+                // If more than 10 times retried, show user warning, but keep reconnecting
+                appEvents.emit(AppEvents.alertWarning, [
+                  'Websocket connection is being disrupted. We keep reconnecting but consider starting new live tailing again.',
+                ]);
```

Comment:
nit: Not sure you're supposed to use appEvents outside of the main app.

Reply:
Yea better to return a data error here and let Grafana show that

Reply:
That would work if we didn't need to retry, in this case we only want to show information to the user but not stop the retries.

Reply:
@hugohaggmark as mentioned on our call I have added console.warn and created an issue for informing user in better way https://github.com/grafana/grafana/issues/29609. As we can't inform the user about issues with their connection, if you think it makes sense to add max retries (e.g. 30?), I can do that.

Reply:
Yeah, I think that is a good idea